Sources/OutlineView/OutlineViewDragAndDrop.swift

```diff
@@ -0,0 +1,139 @@
+    public typealias DataElement = Element
+    
+    public func readPasteboard(item: NSPasteboardItem) -> DraggedItem<Element>? {
+        fatalError()
+    }
+    
+    public func validateDrop(target: DropTarget<Element>) -> ValidationResult<Element> {
+        fatalError()
+    }
+    
+    public func acceptDrop(target: DropTarget<Element>) -> Bool {
+        fatalError()
+    }
+}
+
+public typealias DragSourceWriter<D> = (D) -> NSPasteboardItem?
+public typealias DraggedItem<D> = (item: D, type: NSPasteboard.PasteboardType)
+
+/// An object describing what items are being dragged into an `OutlineView`, and
+/// where in the data heirarchy they are being dropped.
+@available(macOS 10.15, *)
+public struct DropTarget<D> {
+    /// A non-empty array of `DraggedItem` tuples, each with the item
+    /// that is being dragged, and the `NSPasteboard.PasteboardType` that
+    /// generated the item from the dragging pasteboard.
+    public var items: [DraggedItem<D>]
+    
+    /// The `OutlineView` data element into which the target is dropping
+    /// items.
+    ///
+    /// If nil, the items are intended to drop into the root of
+    /// the data heirarchy. Otherwise, they are to be dropped into the given
+    /// item's children array.
+    public var intoElement: D?
+    
+    /// The index of the children array that the dragged items are to be
+    /// dropped.
+    ///
+    /// If this is nil, assume that the items will be dropped into
+    /// a default location for items dropped into `intoElement` (at the end
+    /// of the array, or into a default sorting order). Otherwise, the
+    /// items should be inserted at the given index of the children array.
+    public var childIndex: Int?
+    
+    /// A closure that can be called to determine if the `OutlineView`'s
+    /// representation of a given item is expanded. This may be used in
+    /// `DropReceiver` functions that take a `DropTarget` as a parameter,
+    /// in case the expanded state of the item affects the outcome of the
+    /// function.
+    public var isItemExpanded: ((D) -> Bool)
+}
+
+/// An enum describing the behavior of a dragged item as it moves over
+/// the `OutlineView`.
+public enum ValidationResult<D> {
+    /// Indicates that the dragged item will be copied to the indicated
+    /// location. The given location will be highlighted, and the cursor
+    /// will show a "+" icon.
+    case copy
+    
+    /// Indicates that the dragged item will be moved to the indicated
+    /// location. The given location will be highlighted.
+    case move
+    
+    /// Indicates that the dragged item will not be moved. No location
+    /// will be highlighted.
+    case deny
+    
+    /// Indicates that the dragged item will be copied to a different
+    /// location than it is currently hovering over. The cursor will
+    /// show a "+" icon, and the highlighted location will be determined
+    /// by the bound values in this enum.
```

Comment:
What does "bound values" refer to here?

Reply:
https://github.com/Sameesunkaria/OutlineView/pull/10/commits/0accb88b0d732085942ff36e417ca7d21c0550d9
Looks good!